### DISCUSSION TITLE
Should setAtom be included as a hook dependency?

### DISCUSSION THREAD

**Discussion by tiamed** [pmndrs/jotai#1322] [Q&A] [answered]
I got this warning from eslint, should `setAtom` be included as a hook dependency?

```jsx
const setPreviews = useSetAtom(previewsAtom);

const updatePreviews = useCallback(() => {
  setPreiviews(images)
}, [images]);
```

React Hook useCallback has a missing dependency: 'setPreviews'.
Either include it or remove the dependency array.eslint[react-hooks/exhaustive-deps](https://github.com/facebook/react/issues/14920)

**Comment by dai-shi** [comment, marked as answer]
Yes, because `useSetAtom` can return a different function.
```js
const setPreviews = useSetAtom(cond ? previewsAtom : anotherAtom);

const updatePreviews = useCallback(() => {
  setPreiviews(images)
}, [images, setPreviews]);
```

**Reply by tiamed** [reply]
Is it still needed since there's no conditional situation here?

**Reply by dai-shi** [reply]
You can ignore the warning if you are sure there will be no conditions.

In general, it's recommended to follow the rule. It's not specific to Jotai.

---

From the rule perspective, we don't know if there's a condition or not.
It can come from a custom hook in a different file.

It's the same thing if you use `useCallback` in a custom hook.

```js
const useMyCallback = () => {
  return useCallback(() => 'foo', [])
}

const myCallback = useMyCallback()
useEffect(() => {
  myCallback()
}, [myCallback])
```
We can omit `myCallback` in deps technically, but the rule warns it, because it doesn't know how `useMyCallback` is implemented.

**Reply by tiamed** [reply]
Thanks for your reply.
